### ISSUE TITLE
[QUESTION]Training Loss Much Lower Than Validation Loss in TSMixerModel: Need Help Understanding Why

### ISSUE THREAD

**Issue by erl61** [unit8co/darts#2558]
**Issue**
I am training a TSMixerModel to forecast multivariate time series. The model performs well overall, but I notice that the training loss is consistently much lower than the validation loss (sometimes by orders of magnitude).

I have already tried different loss functions (MAELoss, MapeLoss), and the issue persists. However, when I forecast using this model, I don’t observe signs of overfitting, and the model predictions look good.

**Callback**
I use the following setup for logging the losses:
```
class LossLogger(Callback):
    def __init__(self):
        self.train_loss = []
        self.val_loss = []

    # will automatically be called at the end of each epoch
    def on_train_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
        self.train_loss.append(float(trainer.callback_metrics["train_loss"]))

    def on_validation_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
        if not trainer.sanity_checking:
            self.val_loss.append(float(trainer.callback_metrics["val_loss"]))

loss_logger = LossLogger()
```
**Model**
This is how I initialize the model:

```
progress_bar = TFMProgressBar(enable_sanity_check_bar=False, enable_validation_bar=False)

limit_train_batches = 50
limit_val_batches = 50
max_epochs = 30
batch_size = 64

model_tsm = TSMixerModel(
    input_chunk_length=49,  
        output_chunk_length=130, 
        use_reversible_instance_norm=True,
        optimizer_kwargs={"lr": 1e-4},
        nr_epochs_val_period=1, 
        pl_trainer_kwargs={"gradient_clip_val": 1,
                            "max_epochs": max_epochs,
                            "limit_train_batches": limit_train_batches,
                            "limit_val_batches": limit_val_batches,
                            "accelerator": "auto",
                            "callbacks": [progress_bar, loss_logger]},
        lr_scheduler_cls=torch.optim.lr_scheduler.ExponentialLR,
        lr_scheduler_kwargs={"gamma": 0.999},
        likelihood=QuantileRegression(), 
        loss_fn=None, 
        save_checkpoints=True, 
        force_reset=True,
        batch_size=64,
        random_state=42,
        add_encoders={"cyclic": {"future": ['month', 'day', 'weekday','quarter', 'dayofyear', 'week']}},
        use_static_covariates=True,
        model_name="tsm")
```
**Loss curves**
Here are the plotted loss curves after training:

```
loss_df = pd.DataFrame({'epoch':range(0, len(model_tsm.trainer.callbacks[1].train_loss)),
                        'train_loss':model_tsm.trainer.callbacks[1].train_loss,
                        'val_loss':model_tsm.trainer.callbacks[1].val_loss})

plt.plot(loss_df['epoch'],
         loss_df['train_loss'], color='blue', label='train loss: ' + str(loss_df['train_loss'][-1:].item()))


plt.plot(loss_df['epoch'],
         loss_df['val_loss'], color='orange', label='val loss: ' + str(loss_df['val_loss'][-1:].item()))


plt.gcf().set_size_inches(10, 5)
plt.legend()
plt.show()
```

![image](https://github.com/user-attachments/assets/470f0537-96a9-4d0d-8bb9-351c57ccd636)

**Data**
I create my multivariate time series using from_group_dataframe() as follows:
```
ts_df = TimeSeries.from_group_dataframe(df, group_cols=['group1', 'group2', 'group3'],
                                time_col='ds', value_cols='y', freq='D')
```

**Question**
Why is my training loss significantly lower than the validation loss, sometimes by orders of magnitude? Could it be related to how the data is structured as a list of time series? Is this expected behavior in this scenario, or could there be an issue with scaling or loss calculation?

I appreciate any help or insights!

Thanks!

**Comment by dennisbader** [comment]
Hi @erl61, could you provide a minimal reproducible example including model training (potentially processing of the data), what series you provide to fit and predict.

**Comment by dennisbader** [comment]
The nature of your time series data could indeed be the issue.
I would suggest some thinigs:
- try running training on a small subset of your series and see if it changes something in the loss
- zero to millions can be a very large range, if the value distribution is long-tailed. Let's say that your values in the million range are "outliers" and most of your values are in the range 0-1k, then the scaler would transform most of your values to values close to `0.`, that could indeed mess with model performance. So that would require some data processing (outlier removal, potentially other transformation, ...)
- how many different `"group"` combinations do you have? By default, the `StaticCovariatesTransformer` uses `Ordinal` transformation for categoricals which would assume that your groups have a numeric relationship. If the number of groups is low, you could think of using a OneHotEncoding instead (`transformer_cat`).

**Comment by dennisbader** [comment]
We have the `use_reversible_instance` for our toech models which you could try out (at model creation)
